src/serac/physics/materials/tests/thermomechanical_material.cpp

```diff
@@ -0,0 +1,315 @@
+  auto generalized_fluxes_2 = material.calculateConstitutiveOutputs(
+      displacement_grad_transformed, temperature, temperature_grad, state,
+      displacement_grad_old_transformed, temperature_old, dt);
+  
+  auto [piola_stress, heat_capacity, internal_source, heat_flux] = generalized_fluxes;
+  auto [piola_stress_2, heat_capacity_2, internal_source_2, heat_flux_2] = generalized_fluxes_2;
+  EXPECT_LT(norm(piola_stress - transpose(Q)*piola_stress_2), 1e-12);
+  EXPECT_NEAR(heat_capacity, heat_capacity_2, 1e-12);
+  EXPECT_NEAR(internal_source, internal_source_2, 1e-12);
+  EXPECT_LT(norm(heat_flux - heat_flux_2), 1e-12);
+}
+
+
+TEST(ThermomechanicalMaterial, InternalSourceHasCorrectSign)
+{
+  ThermoelasticMaterial material{.E=100.0, .nu=0.25, .C=1.0, .alpha=1.0e-3, .theta_ref=300.0, .k=1.0};
+  tensor<double, 3, 3> displacement_grad{{{0.35490513, 0.60419905, 0.4275843 },
+                                          {0.23061597, 0.6735498 , 0.43953657},
+                                          {0.25099766, 0.27730572, 0.7678207 }}};
+  double temperature_old = 290.0;
+  tensor<double, 3> temperature_grad{0.87241435, 0.11105156, -0.27708054};
+  ThermoelasticMaterial::State state{};
+  tensor<double, 3, 3> displacement_grad_old{};
+  double temperature = temperature_old;
+  double dt = 1.0;
+  auto [heat_capacity, internal_source, heat_flux] =
+      material.calculateThermalConstitutiveOutputs(displacement_grad, temperature,
+                                                   temperature_grad, state,
+                                                   displacement_grad_old, temperature_old, dt);
+  // should have same sign as sgn(alpha*trEdot), here negative
+  EXPECT_LT(internal_source, 0.0);
+}
+
+TEST(ThermomechanicalMaterial, StressHasCorrectSymmetry)
+{
+  ThermoelasticMaterial material{.E=100.0, .nu=0.25, .C=1.0, .alpha=1.0e-3, .theta_ref=300.0, .k=1.0};
+  tensor<double, 3, 3> displacement_grad{{{0.35490513, 0.60419905, 0.4275843 },
+                                          {0.23061597, 0.6735498 , 0.43953657},
+                                          {0.25099766, 0.27730572, 0.7678207 }}};
+  double temperature_old = 290.0;
+  tensor<double, 3> temperature_grad{0.87241435, 0.11105156, -0.27708054};
+  ThermoelasticMaterial::State state{};
+  tensor<double, 3, 3> displacement_grad_old{};
+  double temperature = temperature_old;
+  double dt = 1.0;
+  auto piola_stress = material.calculateMechanicalConstitutiveOutputs(
+      displacement_grad, temperature, temperature_grad, state,
+      displacement_grad_old, temperature_old, dt);
+  auto deformation_grad = displacement_grad + I;
+  auto kirchhoff_stress = piola_stress*transpose(deformation_grad);
+  double tol = 1e-10;
+  EXPECT_TRUE(is_symmetric(kirchhoff_stress, tol));
+}
+
+
+} // namespace serac
+
+
+int main (int argc, char* argv[])
+{
+  ::testing::InitGoogleTest(&argc, argv);
+
+  int result = RUN_ALL_TESTS();
+
```

Comment:
Does this touch any code that may output via SLIC?  If it does you need to initialize a SLIC logger. Example:

```
  axom::slic::SimpleLogger logger;  // create & initialize test logger, finalized when
                                    // exiting main scope
```

Reply:
I don't know the answer to this. The contents of this file are not used anywhere else, but the includes could have such output.

Reply:
After our discussion in the weekly meeting, I added this line in.